### DISCUSSION TITLE
Search Results v4 - Enable Grouping - Show Collapsed - Does not show null value.

### DISCUSSION THREAD

**Discussion by Abihi23** [microsoft-search/pnp-modern-search#3146] [Ideas]
### Check existing issues before logging new ones
No Issues logged re v4 results grouping.

__Remove the above section before submitting, and removal of the below template will result in the issue being closed.__

**Version used**
Ex: 4.8

**Describe the bug**

Search results v4 - Configured to show documents correctly group by a Property. However, Grouping feature is not  showing the null value. after enabling the Grouping feature.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Go to Page 2 of Search results settings.
2. Click on Grouping Feature, select appropriate Field.
3. Switch Off - Show Collapsed.
4. Make sure that the result contains null values for the selected Field.
5. The null values is not displayed in the search result

**Expected behavior**
I expect that the null values will be displayed on the search result

**Screenshots**
Can be provided if required

**Desktop (please complete the following information):**
 - Edge

**Additional context**

**Comment by wobba** [comment]
Can you explain what you mean by "null" values?

**Comment by bckrma** [comment]
I think i got the same issue with version 4.9.
When you enable the grouping feature for a property which is not filled for all items, a group "null" is created in the detail list.
When you expand this group no items are displayed:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/48834093/4cc3bd1e-5aea-4150-b8ac-4272c73f2a1d)
In the dev tools it looks like this when expanded:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/48834093/98e3c3f0-0199-4e44-91c4-a8233ac7494a)

In my case I group by a managed property (RefinableString00)

**Comment by wobba** [comment]
@bckrma got it.. you could loop over all results and set a value for this property before running it in the grouping perhaps? (haven't looked into it myself). Or we could have a "No value" string localized.

**Comment by Abihi23** [comment]
Many thanks for the feedback!
I mean documents that have no value for this property!
for information, I tested with several properties (not only the RefinableStringXX)

**Comment by bckrma** [comment]
> 

I think a localized "No value" string would be great.
I am not sure how to loop over the results before using it in grouping, or is it possibile to use slots / handlebar expressions within the grouping?

**Comment by wobba** [comment]
I was thinking about handlebars yes, iterate over items in an outer loop, before passing into the grouping control.

**Comment by Anas-Bihi** [comment]
I am coming back to you because after upgrading to the last version 4.14 we have noticed again the same problem with the null value which cannot be expanded. 
Have any corrections been made to this since ?

**Comment by ypcode** [comment]
Hi guys, 

I met the issue too for a customer, and I would like to implement the enhancement/bugfix.
Actually, in our case, the Taxonomy field used for grouping have no values for some items.
It results as a "null" group, but more annoying, the group expand/collapse feature is broken.

I would like to issue a PR for this, but I'd like to confirm the approach with you first.
My approach would be: 
- for Groups on metadata with no value (undefined, or null, or empty string, the title of the group would take its value from a WebPart property that would have a default value. 
That would prevent any kind of breaking change, for existing instances and configurations.

The property would be configurable in the property pane nearby the `Enable grouping`, and `Group by field` settings
with a default value being `No value` (set in the default values of the WebPart manifest file)

I will use this approach in a forked version for my customer temporarily, but we would definitely like to include this in the "official" PnP Search WebPart so we can use the latest and greatest version in the future

Please let me know what you think @wobba , @kasperbolarsen

**Reply by ERTVZ** [reply]
Great suggestion. Personally I would prefer a more generic group title, like '<Empty>' or '<No value>' and all it's localized variants.

**Reply by ypcode** [reply]
Hi @wobba @kasperbolarsen any thoughts about it?

**Reply by kasperbolarsen** [reply]
Sounds awesome.  One small request,  can the No value property be a localized string, same approach as for any other label?

**Reply by ypcode** [reply]
Sure thing! I think it's already the case in the version I forked but I will double check! PR on the way :)

**Reply by ypcode** [reply]
I just checked and finally opted for an empty string by default, I will add the default localized `No value`

**Reply by wobba** [reply]
@ypcode looking forward to the PR :)
